### PR TITLE
fix: Move skip check before call to Youtube-DL

### PR DESCRIPTION
**Pull request by zdimension**
## Description

Before, the skip check was done in `add_media`, i.e. after the call to `process_youtube`. As a result, even for media for which Themerr had already downloaded a theme, Youtube-DL was still called to try and get the stream URL, which 

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: Contents/Code/plex_api_helper.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from plexapi.utils import reverseSearchType
 
 # local imports
-from constants import contributes_to, guid_map
+from constants import contributes_to, guid_map, media_type_dict
```
Comment (ReenigneArcher) [inline Contents/Code/plex_api_helper.py:30]: Need to add this `constants.py`

```py

media_type_dict = dict(
    art=dict(
        method=lambda item: item.uploadArt,
        type='art',
        name='art',
        themerr_data_key='art_url',
        remove_pref='bool_remove_unused_art',
    ),
    posters=dict(
        method=lambda item: item.uploadPoster,
        type='posters',
        name='poster',
        themerr_data_key='poster_url',
        remove_pref='bool_remove_unused_posters',
    ),
    themes=dict(
        method=lambda item: item.uploadTheme,
        type='themes',
        name='theme',
        themerr_data_key='youtube_theme_url',
        remove_pref='bool_remove_unused_theme_songs',
    ),
)
```

I can take care of this if needed.

Comment (zdimension) [reply Contents/Code/plex_api_helper.py:30]: fixed

File: Contents/Code/plex_api_helper.py [hunk 2]
```diff
@@ -260,18 +253,6 @@ def add_media(item, media_type, media_url_id, media_file=None, media_url=None):
...
-                    and themerr_data[media_type_dict[media_type]['themerr_data_key']] == media_url_id:
-                Log.Info('Skipping {} for type: {}, title: {}, rating_key: {}'.format(
-                    media_type_dict[media_type]['name'], item.type, item.title, item.ratingKey
-                ))
-
-                # false because we aren't doing anything, and the listener will not see this item again
-                return False
-        except KeyError:
-            pass
-
```
Comment (ReenigneArcher) [inline Contents/Code/plex_api_helper.py:274]: I think we need to leave this block of code.

File: Contents/Code/plex_api_helper.py [hunk 3]
```diff
@@ -280,15 +274,16 @@ def add_media(item, media_type, media_url_id, media_file=None, media_url=None):
...
         if media_url:
-            uploaded = upload_media(item=item, method=media_type_dict[media_type]['method'], url=media_url)
+            uploaded = upload_media(item=item, method=media_type_dict[media_type]['method'](item), url=media_url)
     else:
         Log.Warning('No theme songs provided for type: {}, title: {}, rating_key: {}'.format(
             item.type, item.title, item.ratingKey
         ))
 
     if uploaded:
+        settings_hash = general_helper.get_themerr_settings_hash()
```
Comment (ReenigneArcher) [inline Contents/Code/plex_api_helper.py:286]:
```suggestion
```

Looks like you added the original `settings_hash` back in, so this isn't needed?

**Pull request merged**